pyflakes/checker.py

```diff
@@ -588,6 +587,17 @@ def _add_to_names(container):
 
 class Scope(dict):
     importStarred = False       # set to True when import * is found
+    # Special key for checking whether a binding is defined only for type checking.
+    TYPE_CHECKING_ONLY = object()
+
+    def __init__(self):
+        super(Scope, self).__init__(self)
+        self[self.TYPE_CHECKING_ONLY] = collections.defaultdict(bool)
+
+    def items(self):
+        for key, val in super(Scope, self).items():
+            if key != self.TYPE_CHECKING_ONLY:
```

Comment:
it hopefully doesn't matter since I don't think this solution is a good one, but when comparing against singletons you'd use `is not` / `is`